v3/src/data-interactive/handlers/attribute-handler.ts

```diff
@@ -1,43 +1,76 @@
-import { isAttributeType } from "../../models/data/attribute"
+import { IAttribute, IAttributeSnapshot, isAttributeType } from "../../models/data/attribute"
+import { IDataSet } from "../../models/data/data-set"
 import { withoutUndo } from "../../models/history/without-undo"
 import { getSharedCaseMetadataFromDataset } from "../../models/shared/shared-data-utils"
 import { t } from "../../utilities/translation/translate"
 import { registerDIHandler } from "../data-interactive-handler"
-import { DIHandler, DIResources, DIValues, diNotImplementedYet } from "../data-interactive-types"
+import { DIHandler, DIResources, DISingleValues, DIValues } from "../data-interactive-types"
 
-function convertAttributeToV2(resources: DIResources) {
+function convertAttributeToV2(attribute: IAttribute, dataContext?: IDataSet) {
+  const metadata = dataContext && getSharedCaseMetadataFromDataset(dataContext)
+  const { name, type, title, description, editable, id, precision } = attribute
+  return {
+    name,
+    type,
+    title,
+    cid: id,
+    // defaultMin: self.defaultMin, // TODO Where should this come from?
+    // defaultMax: self.defaultMax, // TODO Where should this come from?
+    description,
+    // _categoryMap: self.categoryMap, // TODO What is this?
+    // blockDisplayOfEmptyCategories: self.blockDisplayOfEmptyCategories, // TODO What?
+    editable,
+    hidden: (attribute && metadata?.hidden.get(attribute.id)) ?? false,
+    renameable: true, // TODO What should this be?
+    deleteable: true, // TODO What should this be?
+    formula: attribute.formula?.display,
+    // deletedFormula: self.deletedFormula, // TODO What should this be?
+    guid: Number(id), // TODO This is different than v2
+    id: Number(id), // TODO This is different than v2
+    precision,
+    unit: attribute.units
+  }
+}
+
+function convertAttributeToV2FromResources(resources: DIResources) {
   const { attribute, dataContext } = resources
-  const metadata = (dataContext && getSharedCaseMetadataFromDataset(dataContext))
   if (attribute) {
-    const { name, type, title, description, editable, id, precision } = attribute
+    return convertAttributeToV2(attribute, dataContext)
+  }
+}
+
+function convertValuesToAttributeSnapshot(values: DISingleValues): IAttributeSnapshot | undefined {
+  if (values.name) {
+    const userType = values.type && isAttributeType(values.type) ? values.type : undefined
```

Comment:
If you define `isAttributeType()` to return `false` for `undefined`, then this can be simplified to:
```suggestion
    const userType = isAttributeType(values.type) ? values.type : undefined
```